modules/Microsoft.DataFactory/factories/deploy.bicep

```
@@ -155,8 +175,24 @@ resource dataFactory 'Microsoft.DataFactory/factories@2018-06-01' = {
   tags: tags
   identity: identity
   properties: {
-    repoConfiguration: bool(gitConfigureLater) ? null : json('{"type": "${gitRepoType}","accountName": "${gitAccountName}","repositoryName": "${gitRepositoryName}",${((gitRepoType == 'FactoryVSTSConfiguration') ? '"projectName": "${gitProjectName}",' : '')}"collaborationBranch": "${gitCollaborationBranch}","rootFolder": "${gitRootFolder}"}')
-    publicNetworkAccess: bool(publicNetworkAccess) ? 'Enabled' : 'Disabled'
+    repoConfiguration: bool(gitConfigureLater) ? null : union({
+        type: gitRepoType
+        accountName: gitAccountName
+        repositoryName: gitRepositoryName
+        collaborationBranch: gitCollaborationBranch
+        rootFolder: gitRootFolder
+      }, (gitRepoType == 'FactoryVSTSConfiguration' ? {
+        projectName: gitProjectName
+      } : {}), {})
+    publicNetworkAccess: publicNetworkAccess && empty(privateEndpoints) ? 'Enabled' : 'Disabled'
```

If publicNetworkAccess are explicitly set to true, should we disable it anyway if privateendpoints param is not empty?

---

Also, can we avoid to set publicNetworkAccess to true by default? 
I'd suggest to update its declaration above like
```
@description('Optional. Whether or not public endpoint access is allowed for this resource.')
@allowed([
  ''
  'Enabled'
  'Disabled'
])
param publicNetworkAccess string = ''
```
and then set it here with something like
```suggestion
    publicNetworkAccess: !empty(publicNetworkAccess) ? any(publicNetworkAccess) : (!empty(privateEndpoints) ? 'Disabled' : null)
```
as done for other resources

---

> If publicNetworkAccess are explicitly set to true, should we disable it anyway if privateendpoints param is not empty?

Unclear. The user wouldn't actually see that. So the user might be under the impression he set's it to `true` even though we set it to false under the hood. Not clear what our approach should be.

---

> Also, can we avoid to set publicNetworkAccess to true by default? I'd suggest to update its declaration above like
> 
> ```
> @description('Optional. Whether or not public endpoint access is allowed for this resource.')
> @allowed([
>   ''
>   'Enabled'
>   'Disabled'
> ])
> param publicNetworkAccess string = ''
> ```
> 
> and then set it here with something like
> 
> as done for other resources

As per the policy recommendation it should actually be `false` by default. However, I remember we had this exact discussion about another resource and the result was that we did not set it to false.

---

done